File: csharp/src/AutomationFramework.Core/Models/LoginResponse.cs
using System.Text.Json.Serialization;
using AutomationFramework.Core.Support;

namespace AutomationFramework.Core.Models;

/// <summary>
/// A successful login.
/// </summary>
/// <remarks>
/// <c>ToString</c> is overridden so the token cannot reach a log line or an assertion failure message
/// through the record's generated implementation.
/// </remarks>
public sealed record LoginResponse(
    [property: JsonPropertyName("access_token")] string AccessToken,
    [property: JsonPropertyName("token_type")] string TokenType,
    [property: JsonPropertyName("expires_in")] int? ExpiresIn)
{
    /// <summary>The value for an <c>Authorization</c> header.</summary>
    public string BearerHeaderValue() => "Bearer " + AccessToken;

    public override string ToString()
        => $"LoginResponse {{ AccessToken = {Redaction.Marker}, TokenType = {TokenType}, "
            + $"ExpiresIn = {ExpiresIn} }}";
}

File: java/src/main/java/com/company/automation/models/LoginResponse.java
package com.company.automation.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A successful login.
 *
 * <p>{@code toString} is overridden so the token cannot reach a log line or an assertion failure
 * message through the record's generated implementation.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record LoginResponse(
    @JsonProperty("access_token") String accessToken,
    @JsonProperty("token_type") String tokenType,
    @JsonProperty("expires_in") Integer expiresIn) {

  /** The value for an {@code Authorization} header. */
  public String bearerHeaderValue() {
    return "Bearer " + accessToken;
  }

  @Override
  public String toString() {
    return "LoginResponse[accessToken=***REDACTED***, tokenType="
        + tokenType
        + ", expiresIn="
        + expiresIn
        + "]";
  }
}
